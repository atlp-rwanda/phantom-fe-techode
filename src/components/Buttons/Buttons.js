import React from 'react'

export const DangerButton = ({ name ,onclick }) => {
    return ( 
        <button className="w-full p-2 bg-danger-500 rounded text-white shadow hover:bg-danger-400 font-sans font-medium text-1xl " onClick={onclick} >
            {name}
        </button>
     );
}
export const WarningButton = ({ name ,onclick }) => {
    return ( 
        <button className="w-full p-2 bg-warning-100 rounded text-white shadow hover:bg-warning-100 font-sans font-medium text-1xl " onClick={onclick} >
            {name}
        </button>
     );
}
export const SuccessButton = ({ name ,onclick }) => {
    return ( 
        <button className="w-full p-2 bg-green-500 rounded text-white shadow hover:bg-green-400 font-sans font-medium text-1xl " onClick={onclick} >
            {name}
        </button>
     );
}
export const InfoButton = ({ name ,onclick, styles }) => {
    return ( 
        <button className={`p-1 md:px-2 border border-info-600 rounded-md text-info-600 text-xs  bg-info-100 hover:border-info-400 hover:bg-info-200 mr-2 ${styles}`} onClick={onclick} >
            {name}
        </button>
     );
}

export const PrimaryButton = ({ name, onclick }) => {
    return (<button type="submit" onClick={onclick} className="rounded-md bg-primary-600 text-white h-8 md:h-11 w-full md:w-5/6 mt-5 hover:bg-primary-600 mb-10">{name}</button>) 
}
 
export const Primary = ({ name ,onclick , styles }) => {
    return ( 
        <button className={`w-full bg-primary-600 rounded text-white shadow hover:bg-primary-400 font-sans sm:text-sm md:font-medium   text-xs p-1 px-2 ${styles}`} onClick={onclick}  >
            {name}
        </button>
     );
}
export const PermissionButton = ({ name ,onclick , styles, type, svg, alt }) => {
    return ( 
        <button className={` rounded-md text-${type}-600 bg-${type}-200 font-sans text-sm flex items-center  justify-between p-1 my-1  mx-2 ${styles}`} onClick={onclick}  >
            {name}
            <img src={svg} alt={alt} className={`mx-2`} />
        </button>
     );
}


export const ActionLabel = ({ name ,onclick , styles, svg, alt }) => {
    return ( 
        <button className={` rounded-md text-primary-600 bg-primary-200 font-sans text-sm flex items-center  justify-between p-1 my-1  mx-2 ${styles}`} onClick={onclick}  >
            {name}
            <img src={svg} alt={alt} className={`mx-2`} />
        </button>
     );
}

