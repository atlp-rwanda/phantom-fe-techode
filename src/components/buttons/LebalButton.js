import React from 'react';
export const LebalButton = ({ type ,svg, onclick }) => {
    return (        
        <button className={`p-1 md:p-2 border border-${type}-600 rounded-md  bg-${type}-100 hover:border-${type}-400 hover:bg-${type}-200 mr-2`} >
            <img src={svg} alt="Phantom" onClick={onclick} />                                                                        
        </button> 
     );
}
 

const buttonMaker = (color , text) => {
    if(color == 'info' ){
        return(
            <button className={`p-1 md:px-2 border border-info-600 rounded-md text-info-600 text-xs  bg-info-100 hover:border-info-400 hover:bg-info-200 mr-2`} >
                <p> 
                    {text}
                </p>                                                                        
            </button> 
        )
    }
    if(color == 'primary' ){
        return(
            <button className={`p-1 md:px-2 border border-primary-600 rounded-md text-primary-600 text-xs  bg-primary-100 hover:border-primary-400 hover:bg-primary-200 mr-2`} >
                <p> 
                    {text}
                </p>                                                                        
            </button> 
        )
    }
    if(color == 'danger' ){
        return(
            <button className={`p-1 md:px-2 border border-danger-600 rounded-md text-danger-600 text-xs  bg-danger-100 hover:border-danger-400 hover:bg-danger-200 mr-2`} >
                <p> 
                    {text}
                </p>                                                                        
            </button> 
        )
    }
    if(color == 'secondary' ){
        return(
            <button className={`p-1 md:px-2 border border-secondary-600 rounded-md text-secondary-600 text-xs  bg-secondary-100 hover:border-secondary-400 hover:bg-secondary-200 mr-2`} >
                <p> 
                    {text}
                </p>                                                                        
            </button> 
        )
    }
}

export const LebalTextButton = ({ type : colorType , text  }) => {
    const button = buttonMaker(colorType , text);
    return (<button />);
}
 
