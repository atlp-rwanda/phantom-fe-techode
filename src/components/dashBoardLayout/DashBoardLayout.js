import React, { useState } from 'react'
import SideBar from '../sidebar/SideBar';
import DashBoardHeader from '../header/DashBoardHeader';
import '../../assets/style/siderbar.css';
import main from '../../assets/js/main'


const DashBoardLayout = ({ children }) => {
    const [showNav , setShowNav] = useState(true);
    const showNavSecion = () => {
        let show = !showNav;
        setShowNav(show);
    }
    return ( 
        <div className="flex flex-wrap w-screen h-screen overflow-hidden "   style={main.style} >
            {/* =============  Start::left ================= */}
                <div className= {`transition-all ${showNav == true ? ` w-4/12 lg:w-2/12 2xl:w-1/12` :  ` sidebar none-active`}` } >
                    {/*  Start::left side bar */}
                        <SideBar shownav={showNav} show={showNavSecion}  />
                    {/*  End::left side bar */}
                </div>
            {/* ============= Start::left ============= */}

            {/* ============= Start:: Right ============= */}
                <div className={`${showNav == true ? ` w-8/12 lg:w-10/12 2xl:w-11/12` :  `w-screen`}`}>
                  
                    {/* Start:: header  */}
                        <DashBoardHeader shownav={showNav} show={showNavSecion}  /> 
                    {/* End:: Header */}
                    {/* Start:: Content */}
                    <div className="flex flex-wrap max-w-full max-h-full overflow-y-auto ">
                        { children }
                    </div>                        
                    {/* End:: Content */}
                </div>
            {/* ============= End:: Right =============== */}
        </div>
    );
}
 
export default DashBoardLayout;