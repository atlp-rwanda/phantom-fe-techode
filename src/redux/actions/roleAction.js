import  { RoleActions } from "../constants/roleAction"
const { ADD_ROLE,DELETE_PERMISSION, ADD_PERMISSION } = RoleActions ;



export const addRole = (payload) =>{
    return {
        type: ADD_ROLE,
        payload
    }
}
export const deletePermission = (payload) =>{
    return {
        type: DELETE_PERMISSION,
        payload
    }
}

export const addPermission = (payload) =>{
    return {
        type: ADD_PERMISSION,
        payload
    }
}