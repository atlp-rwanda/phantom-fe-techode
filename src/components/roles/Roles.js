import React, { useState, useEffect } from "react";
import { InfoButton, PermissionButton, Primary } from "../buttons/Buttons";
import { LebalButton, LebalTextButton } from "../buttons/LebalButton";
import DashBoardLayout from "../dashBoardLayout/DashBoardLayout";
import Notify from "../../functions/Notify";
import { connect, useSelector } from "react-redux";
import { ToastContainer } from "react-toastify";
import { useDispatch } from "react-redux";

import deletePermissionSVG from "../../assets/svgs/deletePermission.svg";
import close from "../../assets/svgs/close.svg";
import add from "../../assets/svgs/lebals/savePrevelage.svg";
import { addRole } from "../../redux/actions/roleAction";
import { deleteRole } from "../../redux/actions/roleAction";
import { addPermission } from "../../redux/actions/permissionAction";
import { assignPermission } from "../../redux/actions/roleAction";
import { deletePermission } from "../../redux/actions/roleAction";
import deleteIcon from '../../assets/svgs/delete.svg';
import TableRolesSkeleton from "../skeletons/Tables/TableRolesSkeleton";
import TablePermissionSkeleton from "../skeletons/Tables/TablePermissionsSkeleton";



const Roles = (props) => {
  const [loading, setLoading] = useState(true)
  const [addRoleModal, setAddRoleModal] = useState(false);
  const [roleName, setRoleName] = useState("");
  const [addPermissionModal, setAddPermissionModal] = useState(false);
  const [clearPermissionModal, setClearPermissionModal] = useState(false);
  const [permissionName, setPermissionName] = useState("");
  const [assignPermissionM, setAssignPermission] = useState(false);
  const [assignedPermission, setAssignedPermission] = useState("");
  const [role_Id, setRoleId] = useState("");
  const [permissionId, setPermissionId] = useState("");
  const { addRole, addPermission, deleteRole, assignPermission, deletePermission } = props

  useEffect( ()=> {
    setTimeout(()=>{
        setLoading(false)
    }, 3000)
}, [])

  /* ========== Start::  Getting current state ================== */
  let roleCounter = 1;
  let permissionCounter = 1;
  const roles = props.roles;
  const permissions = props.roles
  console.log(permissions);
  /* ============ End::  Getting current state ================== */

  const removeModal = () => {
    let newState = !addRoleModal;
    setAddRoleModal(newState);
  };
  const removePermissionModal = () => {
    let permissionState = !addPermissionModal;
    setAddPermissionModal(permissionState);
  };
  const assignPermissionModal = (role) => {
    setRoleId(role);
    let assignState = !assignPermissionM;
    setAssignPermission(assignState);
  };

  const createRole = (e) => {
    e.preventDefault();

    /* =================================== Start:: validation ================================ */
    if (roleName.trim().length == "") return Notify("please add role", "error");
    /* =================================== End:: validation ================================ */

    let uniqueState = roles.find(element => element.name == roleName);

    if (uniqueState) {
      Notify("Already exist", "error");
    } else {
      addRole(roleName);
       removeModal(); 
    
      Notify("Role has been added", "success");
    }

    setRoleName("");
    
    
  };

  const createPermission = (e) => {
    e.preventDefault();

    /* =================================== Start:: validation ================================ */
    if (permissionName.trim().length == "") return Notify("please add permission", "error");
    /* =================================== End:: validation ================================ */

    addPermission(permissionName);
    setTimeout(() => {
      removePermissionModal();
    }, 5000);
  };

  const assigningPermission = (name) => {
    setAssignedPermission(name);
    Notify(name + " permission selected", "success");
  };
  const assignNewPermission = (e) => {
    e.preventDefault();
    /* =================================== Start:: validation ================================ */
    // if (assignedPermission.trim().length == "")
    //   return Notify("please Choose atleast one permission", "error");
    /* =================================== End:: validation ================================ */
    assignPermission({ role_Id, assignedPermission });
    assignPermissionModal();
    Notify("Permission has been added", "success");
  };

  const removeDeletePermissionModal = (role_Id, permissionId) => {
    let deleteState = !clearPermissionModal;
    setClearPermissionModal(deleteState);
    setPermissionId(permissionId);
    setRoleId(role_Id);
    console.log("PermissionID ", permissionId)
    console.log("Role_ID ", role_Id)
  };

  const removePermission = (e) => {
    e.preventDefault();
    setTimeout(() => {
      removeDeletePermissionModal();
    }, 5000);
  };

  const deleteAssignedPermission = () => {
    // console.log("RoleId " + role_Id + " Permission " + permissionId)
    deletePermission({ Role_Id: role_Id, permissionId });
    setTimeout(() => {
      removeDeletePermissionModal();
    }, 2000);
    Notify("Permission has been deleted", "success");
  }

  return (
    <>
      {/* =========================== Start:: Delete Permission  Model =============================== */}
      <ToastContainer
          position="top-right"
          autoClose={2000}
          hideProgressBar
          newestOnTop={false}
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
          className={`z-50`}
        />
      <div
        className={`h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 ${
          clearPermissionModal === true ? "block" : "hidden"
        }`}
      >        
        <div className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8">
          <div className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  ">
            <h3 className="font-bold text-sm text-center w-11/12 text-danger-500">
              Removing permision
            </h3>
            <hr className=" bg-secondary-150 border my-3 w-full" />
          </div>
          <div className="card-body">
            <h2 className="md:ml-12 text-secondary-500 mt-3 mb-6 text-xs md:text-sm">
              Removing <span className="text-mainColor">Permission's </span>
              Role
            </h2>
            <form
              onSubmit={(e) => removePermission(e)}
              className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
            >
              <div className="w-full flex justify-between">
                <InfoButton
                  name={`Cancel`}
                  onclick={(e) => removeDeletePermissionModal(e.preventDefault())}
                  styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
                />
                <Primary
                  name={`Remove`}
                  styles=" bg-danger-200 hover:bg-danger-100 text-danger-600 py-2 md:w-1/3"
                  type="submit"
                  onclick={deleteAssignedPermission}
                />
              </div>
            </form>
          </div>
        </div>
      </div>
      {/* =========================== End:: Delete Permssion Model =============================== */}

      {/* =========================== Start:: Assign Permission Modal =============================== */}
      <div
        className={`h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 ${
          assignPermissionM === true ? "block" : "hidden"
        }`}
      >
        <div className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8">
          <div className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  ">
            <h3 className="font-bold text-sm text-center w-11/12">
              Assign new Permission
            </h3>
            <div
              className="close-icon w-1/12 cursor-pointer float-right"
              onClick={() => assignPermissionModal()}
            >
              <img src={close} alt="Phantom" className="float-right" />
            </div>
            <hr className=" bg-secondary-150 border my-3 w-full" />
          </div>
          <div className="card-body">
            <h2 className="text-center text-secondary-500 text-xs md:text-sm">
              Assign new permission 
            </h2>
            <div className="mt-3 text-center md:flex md:flex-row">
              {permissions.map((permission) => (
                <PermissionButton
                  key={permission.id}
                  name={permission.name}
                  type="success"
                  styles='bg-success-200 text-success-600 hover:bg-success-100'
                  onclick={() => assigningPermission(permission.name)}
                />
                //   <button
                //   key={permission.id}
                //   type="success"
                //   onClick={() => assigningPermssion(permission.name)}
                // >{permission.name}</button>
              ))}
            </div>
            <div className="w-full flex justify-between md:mt-6">
              <InfoButton
                name={`Cancel`}
                onclick={(e) => assignPermissionModal(e.preventDefault())}
                styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
              />
              <Primary
                name={`Save`}
                styles="py-2 md:w-1/3"
                type="submit"
                onclick={assignNewPermission}
              />
            </div>
          </div>
        </div>
      </div>
      {/* =========================== End:: Assign Permission Modal =============================== */}

      {/* =========================== Start:: Permission Modal =============================== */}
      <div
        className={`h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 ${
          addPermissionModal === true ? "block" : "hidden"
        }`}
      >
        <div className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8">
          <div className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  ">
            <h3 className="font-bold text-sm text-center w-11/12">
              Adding new Permission
            </h3>
            <div
              className="close-icon w-1/12 cursor-pointer float-right"
              onClick={() => removePermissionModal()}
            >
              <img src={close} alt="Phantom" className="float-right" />
            </div>
            <hr className=" bg-secondary-150 border my-3 w-full" />
          </div>
          <div className="card-body">
            <h2 className="md:ml-12 text-secondary-500 text-xs md:text-sm">
              New Permission
            </h2>
            <form
              onSubmit={(e) => createPermission(e)}
              action="/drivers"
              className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
            >
              <div className="input my-3 h-9 ">
                <div className="grouped-input bg-secondary-40 flex items-center  h-full w-full rounded-md">
                  <input
                    type="text"
                    name="permissionName"
                    className=" bg-transparent border-0 outline-none px-5 font-sans text-xs text-secondary-50 h-5 w-4/5"
                    placeholder="Add Permission"
                    value={permissionName}
                    onChange={(e) => setPermissionName(e.target.value)}
                  />
                </div>
              </div>
              <div className="w-full flex justify-between">
                <InfoButton
                  name={`Cancel`}
                  onclick={(e) => removePermissionModal(e.preventDefault())}
                  styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
                />
                <Primary name={`Save`} styles="py-2 md:w-1/3" />
              </div>
            </form>
          </div>
        </div>
      </div>
      {/* =========================== End:: Permission Modal =============================== */}

      {/* =========================== Start:: Role Modal =============================== */}
      <div
        className={`h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 ${
          addRoleModal === true ? "block" : "hidden"
        }`}
      >
        <div className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8">
          <div className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  ">
            <h3 className="font-bold text-sm text-center w-11/12">
              Adding new Role
            </h3>
            <div
              className="close-icon w-1/12 cursor-pointer float-right"
              onClick={() => removeModal()}
            >
              <img src={close} alt="Phantom" className="float-right" />
            </div>
            <hr className=" bg-secondary-150 border my-3 w-full" />
          </div>
          <div className="card-body">
            <h2 className="md:ml-12 text-secondary-500 text-xs md:text-sm">
              New Role
            </h2>
            <form
              onSubmit={(e) => createRole(e)}
              action="/drivers"
              className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
            >
              <div className="input my-3 h-9 ">
                <div className="grouped-input bg-secondary-40 flex items-center  h-full w-full rounded-md">
                  <input
                    type="text"
                    name="roleName"
                    className=" bg-transparent border-0 outline-none px-5 font-sans text-xs text-secondary-50 h-5 w-4/5"
                    placeholder="Add Role"
                    value={roleName}
                    onChange={(e) => setRoleName(e.target.value)}
                  />
                </div>
              </div>
              <div className="w-full flex justify-between">
                <InfoButton
                  name={`Cancel`}
                  onclick={(e) => removeModal(e.preventDefault())}
                  styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
                />
                <Primary name={`Save`} styles="py-2 md:w-1/3" />
              </div>
            </form>
          </div>
        </div>
      </div>
      {/* =========================== End:: Role Modal =============================== */}
      <DashBoardLayout>
        <div className="w-full h-min  lg:w-7/12 bg-white rounded-md p-4 m-2">
          <div className="w-full">
            <div className="card-header flex items-center justify-between">
              <div className="card-title">
                <div className="title mb-3">
                  <h4 className=" text-primary-500 font-bold text-xs md:text-base">
                    List of Roles
                  </h4>
                </div>
              </div>
              <div className="add-new-record">
                <Primary name="New Role" onclick={removeModal} />
              </div>
            </div>
            {/* Skeleton  */}
            {loading && (<TableRolesSkeleton/>)}
                    {!loading && (
                    <>
            <table className="min-w-full border-collapse border-0">
              <thead>
                <tr className="border-b border-b-secondary-100">
                  <th className="text-xs  md:text-md md:font-bold text-mainColor font-sans pt-6 pb-2">
                    #
                  </th>
                  <th className="text-xs  md:text-md md:font-bold text-mainColor font-sans pt-6 pb-2">
                    Role name
                  </th>
                  <th className="text-xs  md:text-md md:font-bold text-mainColor font-sans pt-6 pb-2 md:text-left md:pl-20">
                    Permissions
                  </th>
                </tr>
              </thead>
              <tbody>
                {roles.map((role) => (
                  <tr
                    key={roleCounter}
                    className="h-16 text-right border-b border-b-secondary-100 cursor-pointer hover:bg-gray-100"
                  >
                    <td className="text-secondary-200 font-sans text-xs text-center md:text-sm md:font-sans">
                      {roleCounter++}
                    </td>
                    <td className="text-secondary-200 font-sans text-xs text-center md:text-sm md:font-sans">
                      <LebalTextButton text="J" type="primary" /> {role.name}
                    </td>
                    <td className=" text-secondary-200 font-sans text-xs text-center md:text-sm md:font-sans h-full">
                      <div className="buttons h-full flex flex-col md:flex md:flex-row">
                        {role.permissions.map((permission) => (
                          
                          <PermissionButton
                            key={permission.id}
                            name={permission.permissionName}
                            onclick={() => removeDeletePermissionModal(role.id, permission.id)}
                            type="danger"
                            styles=""
                            svg={deletePermissionSVG}
                          />
                        ))}
                        <PermissionButton
                          name="Add"
                          svg={add}
                          type="success"
                          onclick={() => assignPermissionModal(role.id)}
                        />
                      </div>
                    </td>
                    <td>
                      <div>
                        <LebalButton type={'danger'} svg={deleteIcon} onclick={() => deleteRole(role.id)} />
                      </div>
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
            </>
                    )}
          
          </div>
        </div>
        <div className="w-full h-min lg:w-4/12 bg-white rounded-md m-2 px-4 pt-4">
          <div className="w-full">
            <div className="card-header flex justify-between">
              <div className="card-title">
                <div className="title mb-3">
                  <h4 className=" text-primary-500 font-bold text-xs md:text-base">
                    List of Permissions
                  </h4>
                </div>
              </div>
              <div className="add-new-record">
                {/* <Primary
                  name="New Permission"
                  onclick={removePermissionModal}
                /> */}
              </div>
            </div>
            {loading && (<TablePermissionSkeleton/>)}
                {!loading && (
                <>
            <table className="min-w-full border-collapse border-0">
              <thead>
                <tr className="border-b border-b-secondary-100">
                  <th className="text-xs  md:text-md md:font-bold text-mainColor font-sans pt-6 pb-2">
                    #
                  </th>
                  <th className="text-xs  md:text-md md:font-bold text-mainColor font-sans pt-6 pb-2">
                    Permission name
                  </th>
                </tr>
              </thead>
              <tbody>
                {permissions.map((permission) => (
                  <tr
                    key={permission.id}
                    className="h-16 text-right border-b border-b-secondary-100 cursor-pointer hover:bg-gray-100"
                  >
                    <td className="text-secondary-200 font-sans text-xs text-center md:text-sm md:font-sans">
                      {permissionCounter++}
                    </td>
                    <td className="text-secondary-200 font-sans text-xs text-center md:text-sm md:font-sans">
                      <LebalTextButton text="J" type="primary" />{" "}
                      {permission.name}
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
            </>)}
          </div>
        </div>
      </DashBoardLayout>
    </>
  );
};

const mapToState = (state) => {
 return {
   roles: state.roles,
   permissions: state.permissions
 }
}

export default connect(mapToState, { addRole, addPermission, deleteRole, assignPermission, deletePermission })(Roles);
