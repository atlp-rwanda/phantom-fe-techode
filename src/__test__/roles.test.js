import React from 'react';
import Roles from '../components/roles/Roles';
import DashBoardLayout from '../components/dashBoardLayout/DashBoardLayout';
import { InfoButton, PermissionButton } from '../components/buttons/Buttons';
import { addRole, deleteRole } from '../redux/actions/roleAction';
import store from "../redux/store"
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TableRolesSkeleton from '../components/skeletons/Tables/TableRolesSkeleton';
import { LebalTextButton } from '../components/buttons/LebalButton';
import TablePermissionSkeleton from '../components/skeletons/Tables/TablePermissionsSkeleton';

describe('<Roles />', () =>{
  const props = {
          addRole: jest.fn(),
          deleteRole: jest.fn(),
          assignPermission: jest.fn(),
          deletePermission: jest.fn(),
          roles:[
            {
                id:1,
                name:'Admin',
                permissions:[
                    { id: 1 , permissionName: 'getBus'},
                ]
            }
        ]
      }

      let wrapper;
    beforeEach(()=>{
      wrapper = shallow (<Roles store={store} {...props} /> ).childAt(0).dive();
    });

    it('It should match the snapshot', () => { 
          expect(toJson(wrapper)).toMatchSnapshot();
    });
  
    it('if it should render one TableRolesSkeleton', () =>{
      expect(wrapper.find(TableRolesSkeleton).length).toBe(1);
    });
    
    it('if it should render one DashBoardLayout', () =>{
      expect(wrapper.find(DashBoardLayout).length).toBe(1);
    });

    it('if it should render one TablePermissionSkeleton', () =>{
      expect(wrapper.find(TablePermissionSkeleton).length).toBe(1);
    });
  });