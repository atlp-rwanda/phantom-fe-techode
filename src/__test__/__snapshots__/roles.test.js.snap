// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Roles /> It should match the snapshot 1`] = `
<Fragment>
  <ToastContainer
    autoClose={2000}
    className="z-50"
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={true}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    theme="light"
    transition={[Function]}
  />
  <div
    className="h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm text-center w-11/12 text-danger-500"
        >
          Removing permision
        </h3>
        <hr
          className=" bg-secondary-150 border my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <h2
          className="md:ml-12 text-secondary-500 mt-3 mb-6 text-xs md:text-sm"
        >
          Removing 
          <span
            className="text-mainColor"
          >
            Permission's 
          </span>
          Role
        </h2>
        <form
          className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
          onSubmit={[Function]}
        >
          <div
            className="w-full flex justify-between"
          >
            <InfoButton
              name="Cancel"
              onclick={[Function]}
              styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
            />
            <Primary
              name="Remove"
              onclick={[Function]}
              styles=" bg-danger-200 hover:bg-danger-100 text-danger-600 py-2 md:w-1/3"
              type="submit"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm text-center w-11/12"
        >
          Assign new Permission
        </h3>
        <div
          className="close-icon w-1/12 cursor-pointer float-right"
          onClick={[Function]}
        >
          <img
            alt="Phantom"
            className="float-right"
            src="close.svg"
          />
        </div>
        <hr
          className=" bg-secondary-150 border my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <h2
          className="text-center text-secondary-500 text-xs md:text-sm"
        >
          Assign new permission
        </h2>
        <div
          className="mt-3 text-center md:flex md:flex-row"
        >
          <PermissionButton
            key="1"
            name="Get routes"
            onclick={[Function]}
            styles="bg-success-200 text-success-600 hover:bg-success-100"
            type="success"
          />
          <PermissionButton
            key="2"
            name="Get buses"
            onclick={[Function]}
            styles="bg-success-200 text-success-600 hover:bg-success-100"
            type="success"
          />
          <PermissionButton
            key="3"
            name="Assign buses"
            onclick={[Function]}
            styles="bg-success-200 text-success-600 hover:bg-success-100"
            type="success"
          />
          <PermissionButton
            key="4"
            name="createRoute"
            onclick={[Function]}
            styles="bg-success-200 text-success-600 hover:bg-success-100"
            type="success"
          />
        </div>
        <div
          className="w-full flex justify-between md:mt-6"
        >
          <InfoButton
            name="Cancel"
            onclick={[Function]}
            styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
          />
          <Primary
            name="Save"
            onclick={[Function]}
            styles="py-2 md:w-1/3"
            type="submit"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm text-center w-11/12"
        >
          Adding new Permission
        </h3>
        <div
          className="close-icon w-1/12 cursor-pointer float-right"
          onClick={[Function]}
        >
          <img
            alt="Phantom"
            className="float-right"
            src="close.svg"
          />
        </div>
        <hr
          className=" bg-secondary-150 border my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <h2
          className="md:ml-12 text-secondary-500 text-xs md:text-sm"
        >
          New Permission
        </h2>
        <form
          action="/drivers"
          className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
          onSubmit={[Function]}
        >
          <div
            className="input my-3 h-9 "
          >
            <div
              className="grouped-input bg-secondary-40 flex items-center  h-full w-full rounded-md"
            >
              <input
                className=" bg-transparent border-0 outline-none px-5 font-sans text-xs text-secondary-50 h-5 w-4/5"
                name="permissionName"
                onChange={[Function]}
                placeholder="Add Permission"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <InfoButton
              name="Cancel"
              onclick={[Function]}
              styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
            />
            <Primary
              name="Save"
              styles="py-2 md:w-1/3"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="h-screen w-screen bg-modelColor absolute flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-white w-full  mp:w-8/12  md:w-6/12  xl:w-4/12 2xl:w-3/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full text-mainColor flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm text-center w-11/12"
        >
          Adding new Role
        </h3>
        <div
          className="close-icon w-1/12 cursor-pointer float-right"
          onClick={[Function]}
        >
          <img
            alt="Phantom"
            className="float-right"
            src="close.svg"
          />
        </div>
        <hr
          className=" bg-secondary-150 border my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <h2
          className="md:ml-12 text-secondary-500 text-xs md:text-sm"
        >
          New Role
        </h2>
        <form
          action="/drivers"
          className=" sp:px-8 mp:px-5 sm:px-10  md:px-8 lg:px-12"
          onSubmit={[Function]}
        >
          <div
            className="input my-3 h-9 "
          >
            <div
              className="grouped-input bg-secondary-40 flex items-center  h-full w-full rounded-md"
            >
              <input
                className=" bg-transparent border-0 outline-none px-5 font-sans text-xs text-secondary-50 h-5 w-4/5"
                name="roleName"
                onChange={[Function]}
                placeholder="Add Role"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <InfoButton
              name="Cancel"
              onclick={[Function]}
              styles="py-2 md:w-1/3 bg-primary-200 hover:bg-primary-100 text-primary-500"
            />
            <Primary
              name="Save"
              styles="py-2 md:w-1/3"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <DashBoardLayout>
    <div
      className="w-full h-min  lg:w-7/12 bg-white rounded-md p-4 m-2"
    >
      <div
        className="w-full"
      >
        <div
          className="card-header flex items-center justify-between"
        >
          <div
            className="card-title"
          >
            <div
              className="title mb-3"
            >
              <h4
                className=" text-primary-500 font-bold text-xs md:text-base"
              >
                List of Roles
              </h4>
            </div>
          </div>
          <div
            className="add-new-record"
          >
            <Primary
              name="New Role"
              onclick={[Function]}
            />
          </div>
        </div>
        <TableRolesSkeleton />
      </div>
    </div>
    <div
      className="w-full h-min lg:w-4/12 bg-white rounded-md m-2 px-4 pt-4"
    >
      <div
        className="w-full"
      >
        <div
          className="card-header flex justify-between"
        >
          <div
            className="card-title"
          >
            <div
              className="title mb-3"
            >
              <h4
                className=" text-primary-500 font-bold text-xs md:text-base"
              >
                List of Permissions
              </h4>
            </div>
          </div>
          <div
            className="add-new-record"
          />
        </div>
        <TablePermissionSkeleton />
      </div>
    </div>
  </DashBoardLayout>
</Fragment>
`;
